tiledb/ml/models/base.py

```diff
@@ -60,9 +66,45 @@ def load(self, **kwargs):
         """
 
     @abc.abstractmethod
-    def preview(self, model: typing.Any, **kwargs):
+    def preview(self, **kwargs):
         """
         Abstract method that previews a machine learning model.
         Must be implemented per machine learning framework, i.e, Tensorflow,
         PyTorch etc.
         """
+
+    def set_file_properties(
+        self, framework: str, framework_version: str, preview: str
+    ) -> dict:
+        """
+        Abstract method that returns model array's file properties.
+        :param framework: str. Machine learning framework.
+        :param framework_version: str. Machine learning framework's version.
+        :param preview: str. Machine learning model's preview.
+        :return: dict. Dictionary with model's file properties
+        """
+        return {
+            self.file_properties_obj.ML_FRAMEWORK.value: framework,
+            self.file_properties_obj.STAGE.value: "STAGING",
+            self.file_properties_obj.PYTHON_VERSION.value: platform.python_version(),
+            self.file_properties_obj.FRAMEWORK_VERSION.value: framework_version,
+            self.file_properties_obj.MODEL_PREVIEW.value: preview,
+        }
+
+    @staticmethod
+    def update_model_metadata(array: tiledb.Array, meta: dict):
+        for key, value in meta.items():
+            array.meta[key] = json.dumps(value).encode("utf8")
+
+
+@unique
```

Comment:
Wouldn't be better to be visible before their use? Could you please reformat so `ModelFileProperties ` is defined before `TileDBModel`